dynawo/sources/Modeler/DataInterface/DYNCriteria.cpp

```diff
@@ -0,0 +1,213 @@
+                Criteria(params) {}
+
+LoadCriteria::~LoadCriteria() {}
+
+bool
+LoadCriteria::checkCriteria(bool finalStep) const {
+  if (!finalStep && params_->getScope() == criteria::CriteriaParams::FINAL)
+    return true;
+  double sum = 0.;
+  bool atLeastOneEligibleLoadWasFound = false;
+  for (std::vector<boost::shared_ptr<LoadInterface> >::const_iterator it = loads_.begin(), itEnd = loads_.end();
+      it != itEnd; ++it) {
+    double p = (*it)->getP();
+    if ((params_->hasUMaxPu() || params_->hasUMinPu()) && (*it)->getBusInterface()) {
+      double v = (*it)->getBusInterface()->getStateVarV();
+      double vNom = (*it)->getBusInterface()->getVNom();
+      if (params_->hasUMaxPu() && v > params_->getUMaxPu()*vNom) continue;
+      if (params_->hasUMinPu() && v < params_->getUMinPu()*vNom) continue;
+    }
+    if (params_->getType() == criteria::CriteriaParams::LOCAL_VALUE) {
+      if (params_->hasPMax() && p > params_->getPMax()) {
+        Trace::debug() << DYNLog(SourceAbovePower, (*it)->getID(), p, params_->getPMax(), params_->getId()) << Trace::endline;
+        return false;
+      }
+      if (params_->hasPMin() && p < params_->getPMin()) {
+        Trace::debug() << DYNLog(SourceUnderPower, (*it)->getID(), p, params_->getPMin(), params_->getId()) << Trace::endline;
+        return false;
+      }
+    } else {
+      sum+=p;
+      atLeastOneEligibleLoadWasFound = true;
+    }
+  }
+
+  if (atLeastOneEligibleLoadWasFound && params_->getType() == criteria::CriteriaParams::SUM) {
+    if (params_->hasPMax() && sum > params_->getPMax()) {
+      Trace::debug() << DYNLog(SourcePowerAboveMax, sum, params_->getPMax(), params_->getId()) << Trace::endline;
+      return false;
+    }
+    if (params_->hasPMin() && sum < params_->getPMin()) {
+      Trace::debug() << DYNLog(SourcePowerBelowMin, sum, params_->getPMin(), params_->getId()) << Trace::endline;
+      return false;
+    }
+  }
+
+  return true;
+}
+
+bool
+LoadCriteria::criteriaEligibleForLoad(const boost::shared_ptr<criteria::CriteriaParams>& params) {
+  if (!params->hasPMin() && !params->hasPMax())
+    return false;
+  return true;
+}
+
+void
+LoadCriteria::addLoad(const boost::shared_ptr<LoadInterface>& load) {
+  if (params_->hasUMinNom() &&
+      load->getBusInterface() &&
+      load->getBusInterface()->getVNom() < params_->getUMinNom()) return;
+  if (params_->hasUMaxNom() &&
+      load->getBusInterface() &&
+      load->getBusInterface()->getVNom() > params_->getUMaxNom()) return;
+  loads_.push_back(load);
+}
+
+GeneratorCriteria::GeneratorCriteria(const boost::shared_ptr<criteria::CriteriaParams>& params) :
+                Criteria(params) {}
+
+GeneratorCriteria::~GeneratorCriteria() {}
+
+bool
+GeneratorCriteria::checkCriteria(bool finalStep) const {
+  if (!finalStep && params_->getScope() == criteria::CriteriaParams::FINAL)
+    return true;
+  double sum = 0.;
+  bool atLeastOneEligibleLoadWasFound = false;
```

Comment:
Change to atLeastOneEligibleGeneratorWasFound

Reply:
Solved